domains/Vault/ManageTasks/Web/ViewHelpers/VaultTasksIndexViewHelper.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Vault\ManageTasks\Web\ViewHelpers;
+
+use App\Helpers\DateHelper;
+use App\Models\User;
+use App\Models\Vault;
+use Carbon\Carbon;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+
+class VaultTasksIndexViewHelper
+{
+    public static function data(Vault $vault, User $user): Collection
+    {
+        $contacts = $vault->contacts()
+            ->orderBy('last_name', 'asc')
+            ->get();
+
+        $contactsCollection = collect();
+        foreach ($contacts as $contact) {
+            $tasks = DB::table('contact_tasks')
```

Comment:
Isn't there a solution where you don't trigger this query on each contact? 1 query for all contacts should be possible ...

Reply:
The thing is:

1- the tasks are displayed per contact on this page, so I need a way to check all the contacts that have tasks.
2- but the current way I've done it, is not optimized at all, I know, because I don't know how to say: give me all the contacts who have at least one uncompleted task.

I need to think.